tests/test.py

```diff
@@ -750,23 +750,53 @@ def testReadAcrastructInAcrastruct(self):
         self.assertNotEqual(row['data'][fake_offset:].decode('ascii', errors='ignore'),
                             row['raw_data'])
 
-
-class CensorTest(BaseTestCase):
+class BaseCensorTest(BaseTestCase):
+    CENSOR_CONFIG_FILE = 'default.yaml'
     def fork_acra(self, popen_kwargs: dict=None, **acra_kwargs: dict):
-        acra_kwargs['acracensor_config_file'] = 'tests/acra-censor_configs/acra-censor-1.yaml'
+        acra_kwargs['acracensor_config_file'] = self.CENSOR_CONFIG_FILE
         return self._fork_acra(acra_kwargs, popen_kwargs)
 
-    def testCensorWithMysql(self):
-        with self.assertRaises(sa.exc.OperationalError):
-            row_id = self.get_random_id()
-            result = self.engine1.execute(
-                sa.select([test_table])
-                    .where(test_table.c.id == row_id))
-            result.fetchone()
 
-    def testCensorWithPostgres(self):
-        #stub test
-        pass
+class CensorBlacklistTest(BaseCensorTest):
+    CENSOR_CONFIG_FILE = 'tests/acra-censor_configs/acra-censor_blacklist.yaml'
+    def testBlacklist(self):
+        if TEST_MYSQL:
+            with self.assertRaises(sa.exc.OperationalError):
+                result = self.engine1.execute(sa.text("select data from test where id='1'"))
+
+            with self.assertRaises(sa.exc.OperationalError):
+                result = self.engine1.execute(sa.text("select data_raw from test"))
+
+            with self.assertRaises(sa.exc.OperationalError):
+                result = self.engine1.execute(sa.text("select * from acrarollback_output"))
+
+        if TEST_POSTGRESQL:
+            with self.assertRaises(sa.exc.ProgrammingError):
+                result = self.engine1.execute(sa.text("select data from test where id='1'"))
+
+            with self.assertRaises(sa.exc.ProgrammingError):
+                result = self.engine1.execute(sa.text("select data_raw from test"))
+
+            with self.assertRaises(sa.exc.ProgrammingError):
+                result = self.engine1.execute(sa.text("select * from acrarollback_output"))
+
+class CensorWhitelistTest(BaseCensorTest):
+    CENSOR_CONFIG_FILE = 'tests/acra-censor_configs/acra-censor_whitelist.yaml'
+    def testWhitelist(self):
+        if TEST_MYSQL:
+            with self.assertRaises(sa.exc.OperationalError):
```

Comment:
to avoid duplicated queries, we can do something like:
```
expectedException = None
if TEST_MYSQL:
  expectedException = sa.exc.OperationalError
else if TEST_POSTGRESQL:
  expectedException = sa.exc.ProgrammingError
with self.assertRaises(expectedException):
  ...
```